rs/cli/src/auth.rs

```diff
@@ -158,6 +318,7 @@ impl Auth {
         }
     }
 
+    /// If it is called it is expected to retrieve an Auth of type Hsm or fail
```

Comment:
Perfect.  I was dying to get to this point in the refinement of this function.

Reply:
Was an obvious choice after the refactor